### PR TITLE
[ApiResponseException] set previous exception

### LINKED ISSUE #277: ApiResponseException should have a getter for the original Guzzle RequestException.
I've just been experiencing some connectivity issues on a client's server and noticed that the Zendesk ApiResponseException does not provide for a way to access the original response exception object, which means I have no way to write clean error handling code for errors that aren't related to Zendesk or its servers.

You already receive the Guzzle RequestException in the `__construct()`, it would be quite straightforward to just store it in a variable and make it accessible like you do with `getErrorDetails()`.

Happy to write a PR for this if it's something you'd be open to merging!

### PR DESCRIPTION
**Pull request by kcasas**
/cc @zendesk/mintegrations

### Description

Added previous exception when calling parent constructor on ApiResponseException. Previous exceptions could be retrieved by calling `$e->getPrevious()`

### References
* Issue: https://github.com/zendesk/zendesk_api_client_php/issues/277

### Risks
* none

### CODE REVIEW

File: tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+        $mockException->method('hasResponse')->willReturn(true);
+        $apiException = new ApiResponseException($mockException);
+        $previousException = $apiException->getPrevious();
+
+        $this->assertEquals($mockException, $previousException, $message);
+    }
+}
```
Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php]: Add an empty line here.

File: tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+use Zendesk\API\UnitTests\BasicTest;
+use Zendesk\API\Exceptions\ApiResponseException;
+
+class ExceptionsTest extends BasicTest
+{
+
+    public function testPreviousException()
+    {
+        $message = 'test if previous exception was correctly set on ApiResponseException';
```
Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php]: The message could be improved to specify what actually failed in the test, something like - "The previous exception was **not** passed to ApiResponseException".

File: tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?php
```
Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php]: There are some code formatting issues on this class, hence the failed travis test. You can fix them by using phpcbf and setting the standard to PSR2 (`vendor/bin/phpcbf --standard=PSR2`).

File: tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Zendesk\API\UnitTests\Utilities;
```
Comment (samgavinio) [inline tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php]: I think this shouldn't this live under the `Zendesk\API\UnitTests\Utilities` namespace. Perhaps `Zendesk\API\UnitTests\Exceptions` would be better.

File: tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+use Zendesk\API\UnitTests\BasicTest;
+use Zendesk\API\Exceptions\ApiResponseException;
+
+class ExceptionsTest extends BasicTest
+{
+
+    public function testPreviousException()
```
Comment (miogalang) [inline tests/Zendesk/API/UnitTests/Utilities/ExceptionsTest.php]: Please add a docblock and describe what this tests for

File: tests/Zendesk/API/UnitTests/Exceptions/ApiResponseExceptionTest.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+     * Tests if previous exception was passed to ApiResponseException
+     */
+    public function testPreviousException()
+    {
+        $message = 'The previous exception was not passed to ApiResponseException';
+        $mockException = $this
+            ->getMockBuilder('GuzzleHttp\Exception\RequestException')
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
```
Comment (miogalang) [inline tests/Zendesk/API/UnitTests/Exceptions/ApiResponseExceptionTest.php]: why is the `;` on a new line?

Comment (kcasas) [inline tests/Zendesk/API/UnitTests/Exceptions/ApiResponseExceptionTest.php]: I find it more convenient to add new function calls in the chain when it is formatted this way but I'll change it anyway.

**Comment by kcasas** [comment]
:haircut_man: @joseconsador @miogalang

**Pull request merged**